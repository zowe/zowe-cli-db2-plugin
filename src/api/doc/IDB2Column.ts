/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Column description
 * @interface IDB2Column
 * @export
 */
export type IDB2Column = {
    /**
     * The catalog name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_CAT: string;

    /**
     * The scheme name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_SCHEM: string;

    /**
     * The table name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_NAME: string;

    /**
     * The column name
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_NAME: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    DATA_TYPE: number;

    /**
     * The string representation of the SQL data type
     * @type {string}
     * @memberof IDB2Column
     */
    TYPE_NAME: string;

    /**
     * The length of the column
     * @type {number}
     * @memberof IDB2Column
     */
    COLUMN_SIZE: number;

    /**
     * Buffer length needed to store data
     * @type {number}
     * @memberof IDB2Column
     */
    BUFFER_LENGTH: number;

    /**
     * Number of decimal digits for a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    DECIMAL_DIGITS: number | null;

    /**
     * The radix of a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    NUM_PREC_RADIX: number | null;

    /**
     * Indicates whether the column can contain the null value
     *
     * 0 - NULLs are not allowed
     * 1 - NULLs are allowed
     * @type {number}
     * @memberof IDB2Column
     */
    NULLABLE: number;

    /**
     * Description of the column
     * @type {string}
     * @memberof IDB2Column
     */
    REMARKS: string;

    /**
     * The default value
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_DEF: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    SQL_DATA_TYPE: number;

    /**
     * The subtype of a datetime type
     *
     * 1 - Date
     * 2 - Time
     * 3 - Timestamp
     * @type {number|null}
     * @memberof IDB2Column
     */
    SQL_DATETIME_SUB: number | null;

    /**
     * Length in bytes of the column
     * @type {number|null}
     * @memberof IDB2Column
     */
    CHAR_OCTET_LENGTH: number | null;

    /**
     * Position of the column in the table starting from 1
     * @type {number}
     * @memberof IDB2Column
     */
    ORDINAL_POSITION: number;

    /**
     * Indicates whether the column can contain the null value
     *
     * NO - NULLs are not allowed
     * YES - NULLs are allowed
     * @type {string}
     * @memberof IDB2Column
     */
    IS_NULLABLE: string;
};
