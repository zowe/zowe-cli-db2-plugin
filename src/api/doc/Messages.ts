/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { IMessageDefinition, apiErrorHeader } from "@zowe/imperative";

export const noDB2Input: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 parameters were supplied.",
};

export const noHostName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server hostname was supplied.",
};

export const noPortNumber: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server port number was supplied.",
};

export const noUserName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 User ID was supplied.",
};

export const noPassword: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 user password was supplied.",
};

export const noDatabaseName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 database name was supplied.",
};

export const noTableName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 table name was supplied.",
};
