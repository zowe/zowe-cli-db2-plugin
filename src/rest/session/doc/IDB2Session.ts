/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface for DB2 connection parameters
 * @export
 * @interface
 */
export interface IDB2Session {

    /**
     * The name or IP address of the server to connect to
     * @type {string}
     * @memberof IDB2Session
     */
    hostname: string;

    /**
     * The port number where the server is listening
     * @type {number}
     * @memberof IDB2Session
     */
    port: number;

    /**
     * The user ID
     * @type {string}
     * @memberof IDB2Session
     */
    user: string;

    /**
     * The user's password
     * @type {string}
     * @memberof IDB2Session
     */
    password: string;

    /**
     * The database name to use
     * @type {string}
     * @memberof IDB2Session
     */
    database: string;

    /**
     * The path to a SSL Certificate file from a server or a CA signed certificate
     *
     * When specified the connection to the database server will be done over SSL.
     * @type {string}
     * @memberof IDB2Session
     */
    sslFile?: string;
}
