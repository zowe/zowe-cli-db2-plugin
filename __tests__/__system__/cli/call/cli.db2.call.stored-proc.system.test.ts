/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";

let TEST_ENV: ITestEnvironment<ITestPropertiesSchema>;


describe("db2 call stored procedure command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENV = await TestEnvironment.setUp({
            installPlugin: true,
            tempProfileTypes: ["db2"],
            testName: "call_sp_command"
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/sp_help.sh", TEST_ENV);
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
    });
});
