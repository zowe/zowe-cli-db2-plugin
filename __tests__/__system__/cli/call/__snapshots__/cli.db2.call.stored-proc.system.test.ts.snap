// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db2 call stored procedure command should display the help 1`] = `
"
 COMMAND NAME
 ------------

   procedure | proc | sp

 DESCRIPTION
 -----------

   Call a Db2 stored procedure. Specify the stored procedure name and optionally
   provide values.

 USAGE
 -----

   zowe db2 call procedure <routine> [options]

 POSITIONAL ARGUMENTS
 --------------------

   routine		 (string)

      The name of a Db2 stored procedure

 OPTIONS
 -------

   --parameters  | -p (array)

      Values to bind to the stored procedure parameters

 DB2 CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The Db2 server host name

   --port  | -P (number)

      The Db2 server port number

   --user  | -u (string)

      The Db2 user ID (may be the same as the TSO login)

   --password  | --pass | --pw (string)

      The Db2 password (may be the same as the TSO password)

   --database  | --db (string)

      The name of the database

   --sslFile  | --ssl (string)

      Path to an SSL Certificate file

 PROFILE OPTIONS
 ---------------

   --db2-profile  | --db2-p (string)

      The name of a (db2) profile to load for this command execution.

   --base-profile  | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type  | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value  | --tv (string)

      The value of the token to pass to the API.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Call stored procedure DEMO.SP1:

      $ zowe db2 call procedure \\"DEMO.SP1\\"

   - Call a stored procedure and pass values for parameter
   indicators:

      $ zowe db2 call procedure \\"DEMO.SP2(?, ?)\\" --parameters \\"Hello\\" \\"world!\\"

   - Call a stored procedure and pass values for two output
   parameters. The first output requires a 2-character buffer. The second output is
   a message that will be truncated to the length of the placeholder.:

      $ zowe db2 call procedure \\"DEMO.SP3(NULL, ?, ?)\\" --parameters \\"00\\" \\"message_placeholder_message_placeholder\\"

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: procedure.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   procedure | proc | sp\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Call a Db2 stored procedure. Specify the stored procedure name and optionally\\\\n   provide values.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 call procedure <routine> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   routine\\\\t\\\\t (string)\\\\n\\\\n      The name of a Db2 stored procedure\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --parameters  | -p (array)\\\\n\\\\n      Values to bind to the stored procedure parameters\\\\n\\\\n DB2 CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The Db2 server host name\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The Db2 server port number\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The Db2 user ID (may be the same as the TSO login)\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      The Db2 password (may be the same as the TSO password)\\\\n\\\\n   --database  | --db (string)\\\\n\\\\n      The name of the database\\\\n\\\\n   --sslFile  | --ssl (string)\\\\n\\\\n      Path to an SSL Certificate file\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --db2-profile  | --db2-p (string)\\\\n\\\\n      The name of a (db2) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Call stored procedure DEMO.SP1:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP1\\\\\\"\\\\n\\\\n   - Call a stored procedure and pass values for parameter\\\\n   indicators:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP2(?, ?)\\\\\\" --parameters \\\\\\"Hello\\\\\\" \\\\\\"world!\\\\\\"\\\\n\\\\n   - Call a stored procedure and pass values for two output\\\\n   parameters. The first output requires a 2-character buffer. The second output is\\\\n   a message that will be truncated to the length of the placeholder.:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP3(NULL, ?, ?)\\\\\\" --parameters \\\\\\"00\\\\\\" \\\\\\"message_placeholder_message_placeholder\\\\\\"\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   procedure | proc | sp\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Call a Db2 stored procedure. Specify the stored procedure name and optionally\\\\n   provide values.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 call procedure <routine> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   routine\\\\t\\\\t (string)\\\\n\\\\n      The name of a Db2 stored procedure\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --parameters  | -p (array)\\\\n\\\\n      Values to bind to the stored procedure parameters\\\\n\\\\n DB2 CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The Db2 server host name\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The Db2 server port number\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The Db2 user ID (may be the same as the TSO login)\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      The Db2 password (may be the same as the TSO password)\\\\n\\\\n   --database  | --db (string)\\\\n\\\\n      The name of the database\\\\n\\\\n   --sslFile  | --ssl (string)\\\\n\\\\n      Path to an SSL Certificate file\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --db2-profile  | --db2-p (string)\\\\n\\\\n      The name of a (db2) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Call stored procedure DEMO.SP1:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP1\\\\\\"\\\\n\\\\n   - Call a stored procedure and pass values for parameter\\\\n   indicators:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP2(?, ?)\\\\\\" --parameters \\\\\\"Hello\\\\\\" \\\\\\"world!\\\\\\"\\\\n\\\\n   - Call a stored procedure and pass values for two output\\\\n   parameters. The first output requires a 2-character buffer. The second output is\\\\n   a message that will be truncated to the length of the placeholder.:\\\\n\\\\n      $ zowe db2 call procedure \\\\\\"DEMO.SP3(NULL, ?, ?)\\\\\\" --parameters \\\\\\"00\\\\\\" \\\\\\"message_placeholder_message_placeholder\\\\\\"\\\\n\\\\n\\"
}"
`;
